### ISSUE TITLE
exec-builtin should also warn on references to the exec function

### ISSUE THREAD

**Issue by JelleZijlstra** [astral-sh/ruff#28011]
### Summary

Consider this code:

```python
exec("hi")  # exec-builtin

list(map(exec, ["hi"]))

eval("hi")  # suspicious-eval-usage

list(map(eval, ["hi"]))  # suspicious-eval-usage
```

Ruff (in preview mode) currently flags the indirect use of `eval` through `suspicious-eval-usage` / S307, but the analogous rule for `exec` (`exec-builtin` / S102) only flags the call, not mere use of the builtin. Both should be consistent.

Aside: the two lint rules probably should also have names that are more similar, since they do pretty much the same thing.
